qiskit/algorithms/eigen_solvers/numpy_eigen_solver.py

```diff
@@ -193,8 +193,67 @@ def _eval_aux_operators(
             values[key] = (value, 0.0)
         return values
 
+    @staticmethod
+    def _eval_transition_amplitudes(
+        aux_operators: ListOrDict[OperatorBase], wavefn, wavefm, threshold: float = 1e-12
+    ) -> ListOrDict[Tuple[complex, complex]]:
+        """Evaluate the transition amplitudes for the states n and m and the list of auxiliaries.
+
+        The formula we use here is a simpler version of Eq. 13 in
+        https://doi.org/10.1103%2Fphysrevresearch.3.023244. It is defined only for n!=m.
+        If n=m, then the _eval_aux_operators must be used instead.
+
+        """
+
+        values: ListOrDict[Tuple[complex, complex]]
+
+        # As a list, aux_operators can contain None operators for which None values are returned.
+        # As a dict, the None operators in aux_operators have been dropped in compute_eigenvalues.
+        if isinstance(aux_operators, list):
+            values = [None] * len(aux_operators)
+            key_op_iterator = enumerate(aux_operators)
+        else:
+            values = {}
+            key_op_iterator = aux_operators.items()
+        for key, operator in key_op_iterator:
+            if operator is None:
+                continue
+            value = 0.0
+            if operator.coeff != 0:
+                mat = operator.to_spmatrix()
+                # Terra doesn't support sparse yet, so do the matmul directly if so
```

Comment:
I'm not sure I fully understand this. `operator.to_spmatrix()` is a Terra method, is it not? If that's the case, where does Terra not support sparse matrices yet? Or do you mean in the opflow parts being used next?

Reply:
I based my _`eval_transition_amplitudes` method on the logic of the existing `_eval_aux_operators`. I don't know if the support of sparse matrix has been implemented since, but I treated this sparsity as it was done in the other method.